data_api/snapshot_generator/src/test/scala/uk/ac/wellcome/platform/snapshot_generator/services/SnapshotServiceTest.scala

```diff
@@ -0,0 +1,369 @@
+            downloadFile)
+
+          val contents = readGzipFile(downloadFile.getPath)
+          val expectedContents = visibleWorks
+            .map {
+              DisplayWorkV2(_, includes = AllWorksIncludes())
+            }
+            .map {
+              mapper.writeValueAsString(_)
+            }
+            .mkString("\n") + "\n"
+
+          contents shouldBe expectedContents
+
+          result shouldBe CompletedSnapshotJob(
+            snapshotJob = snapshotJob,
+            targetLocation =
+              s"http://localhost:33333/${publicBucket.name}/$publicObjectKey"
+          )
+        }
+    }
+
+  }
+
+  it("completes a very large snapshot generation successfully") {
+    withFixtures {
+      case (snapshotService: SnapshotService, indexNameV1, _, publicBucket) =>
+        // Create a collection of works.  The use of Random is meant
+        // to increase the entropy of works, and thus the degree to
+        // which they can be gzip-compressed
+        val works = (1 to 5000).map { version =>
+          IdentifiedWork(
+            canonicalId = Random.alphanumeric.take(7).mkString,
+            title = Some(Random.alphanumeric.take(1500).mkString),
+            sourceIdentifier = SourceIdentifier(
+              identifierScheme = IdentifierSchemes.miroImageNumber,
+              ontologyType = itemType,
+              value = Random.alphanumeric.take(10).mkString
+            ),
+            description = Some(Random.alphanumeric.take(2500).mkString),
+            publicationDate = Some(Period(label = version.toString)),
+            version = version
+          )
+        }
+
+        insertIntoElasticsearch(indexNameV1, itemType, works: _*)
+
+        val publicObjectKey = "target.txt.gz"
+        val snapshotJob = SnapshotJob(
+          publicBucketName = publicBucket.name,
+          publicObjectKey = publicObjectKey,
+          apiVersion = ApiVersions.v1
+        )
+
+        val future = snapshotService.generateSnapshot(snapshotJob)
+
+        whenReady(future) { result =>
+          val downloadFile =
+            File.createTempFile("snapshotServiceTest", ".txt.gz")
+          s3Client.getObject(
+            new GetObjectRequest(publicBucket.name, publicObjectKey),
+            downloadFile)
+
+          val contents = readGzipFile(downloadFile.getPath)
+          val expectedContents = works
+            .map {
+              DisplayWorkV1(_, includes = AllWorksIncludes())
+            }
+            .map {
+              mapper.writeValueAsString(_)
+            }
+            .mkString("\n") + "\n"
+
+          contents shouldBe expectedContents
+
+          result shouldBe CompletedSnapshotJob(
+            snapshotJob = snapshotJob,
+            targetLocation =
+              s"http://localhost:33333/${publicBucket.name}/$publicObjectKey"
+          )
+        }
+    }
+  }
+
+  ignore(
+    "succeeds a snapshot generation even if one of the items in the index is invalid") {
```

Comment:
I actually I'm starting to think that it should crash with invalid works and this is not the behaviour that we want. Thoughts?